File: go/sprint1_nonfinals/i.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func isPowerOfFour(number int) bool {
	// Ваше решение
}

func main() {
	scanner := makeScanner()
	number := readInt(scanner)
	if isPowerOfFour(number) {
		fmt.Println("True")
	} else {
		fmt.Println("False")
	}
}

func makeScanner() *bufio.Scanner {
	const maxCapacity = 3 * 1024 * 1024
	buf := make([]byte, maxCapacity)
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Buffer(buf, maxCapacity)
	return scanner
}

func readInt(scanner *bufio.Scanner) int {
	scanner.Scan()
	stringInt := scanner.Text()
	res, _ := strconv.Atoi(stringInt)
	return res
}

File: cpp/sprint1_nonfinals/i.cpp
#include <iostream>

using namespace std;

bool IsPowerOfFour(int number) {
	// Здесь реализация вашего решения

}

void print(bool result) {
	if (result)
		cout << "True";
	else
		cout << "False";
	cout << endl;
}

int main() {
	int number;
	cin >> number;
	print(IsPowerOfFour(number));
}
